### PR TITLE
Improve dashboard accessibility

### PR DESCRIPTION
**Pull request by dulcimerist**
## Issue
Resolves #79

## What was done
- [x] Cleaned up a number of buttons that were actually divs with lots of JS hand-rolling
- [x] Added aria labels, descriptions, etc... to the dashboard components
- [x] Rewrote logic to tab-select options in the main menu dropdown and automatically close the dropdown on blur
- [x] Added styling to improve contrast on selected elements
- [x] Tied text field to its constraint description for screen readers
 
## How to test
1. Navigate the dashboard with only the keyboard
2. Turn off monitor and try to navigate dashboard with VoiceOver

### CODE REVIEW

File: ui/src/SpaceDashboard/SpaceForm.tsx [hunk 1]
```diff
@@ -37,6 +37,12 @@ function SpaceForm({
 }: SpaceFormProps): JSX.Element {
     const maxLength = 40;
     const [formSpace, setFormSpace] = useState<Space>(initializeSpace());
+    const spaceNameInputRef = createRef<HTMLInputElement>();
+
+    useEffect(() => {
+        console.log(spaceNameInputRef);
```
Comment (aclift19Ford) [inline ui/src/SpaceDashboard/SpaceForm.tsx:43]: We left in a console log.....

**Review by smle225 — CHANGES_REQUESTED** [review]
Functionality works great. But Create button is little off
Screen Shot 2021-01-21 at 11 51 35 AM

File: ui/src/SpaceDashboard/SpaceDashboard.tsx [hunk 2]
```diff
@@ -107,12 +107,14 @@ function SpaceDashboard({
 
     function NewSpaceButton(): JSX.Element {
         return (
-            <button className="createNewSpaceButton" onClick={onCreateNewSpaceButtonClicked}>
-                <i className="material-icons createNewSpaceIcon">
-                    add_circle_outline
-                </i>
-                Create New Space
-            </button>
+            <div>
```
Comment (aclift19Ford) [inline ui/src/SpaceDashboard/SpaceDashboard.tsx:110]: What was the intention of wrapping this in a div? I'm curious.

File: ui/src/SpaceDashboard/SpaceDashboard.scss [hunk 3]
```diff
@@ -34,6 +34,13 @@ $spaceTileWidth: 400px;
   grid-gap: 32px;
   margin-top: 50px;
   padding: 88px;
+
+  & > * {
```
Comment (aclift19Ford) [inline ui/src/SpaceDashboard/SpaceDashboard.scss:38]: Also curious what this is for

File: ui/src/SpaceDashboard/SpaceDashboard.tsx (hunk 2 above)
Comment (dulcimerist) [reply ui/src/SpaceDashboard/SpaceDashboard.tsx:110]: I believe the idea was to bring consistency to the new space button and the space tiles - @dctid, you had more of an opinion on this than I when we were pairing; I don't want to misrepresent your reasoning

File: ui/src/SpaceDashboard/SpaceDashboard.scss (hunk 3 above)
Comment (dulcimerist) [reply ui/src/SpaceDashboard/SpaceDashboard.scss:38]: We got additional feedback from Sharon where she pointed out that the styling that you and I added to the space tiles wasn't consistently applied to the create new space button; this forces any tile added to the grid to all be the same position and size.

**Review by smle225 — APPROVED** [review]

**Pull request merged**